tests/matrix_free/matrix_vector_rt_face_common.h

```diff
@@ -0,0 +1,301 @@
+             const Vector<Number> &                       src,
+             const std::pair<unsigned int, unsigned int> &face_range) const
+  {
+    FEFaceEvaluation<dim, fe_degree, n_q_points_1d, dim, Number> fe_eval(data,
+                                                                         true);
+    FEFaceEvaluation<dim, fe_degree, n_q_points_1d, dim, Number> fe_eval_n(
+      data, false);
+
+    unsigned int n_cells =
+      data.get_dof_handler().get_triangulation().n_active_cells();
+    Number piola =
+      (dim == 2) ? n_cells : Utilities::pow((int)std::cbrt(n_cells), 4);
+
+    for (unsigned int face = face_range.first; face < face_range.second; ++face)
+      {
+        fe_eval.reinit(face);
+        fe_eval_n.reinit(face);
+
+        fe_eval.gather_evaluate(src,
+                                EvaluationFlags::values |
+                                  EvaluationFlags::gradients);
+        fe_eval_n.gather_evaluate(src,
+                                  EvaluationFlags::values |
+                                    EvaluationFlags::gradients);
+
+        for (unsigned int q = 0; q < fe_eval.n_q_points; ++q)
+          {
+            fe_eval.submit_value(Number(10. * piola) * fe_eval.get_value(q), q);
+            fe_eval_n.submit_value(Number(10. * piola) * fe_eval_n.get_value(q),
+                                   q);
+
+            fe_eval.submit_gradient(Number(piola) * fe_eval.get_gradient(q), q);
+            fe_eval_n.submit_gradient(Number(piola) * fe_eval_n.get_gradient(q),
+                                      q);
+          }
+
+        fe_eval.integrate_scatter(EvaluationFlags::values |
+                                    EvaluationFlags::gradients,
+                                  dst);
+        fe_eval_n.integrate_scatter(EvaluationFlags::values |
+                                      EvaluationFlags::gradients,
+                                    dst);
+      }
+  };
+
+  void
+  operator_boundary(
+    const MatrixFree<dim, Number> &              data,
+    Vector<Number> &                             dst,
+    const Vector<Number> &                       src,
+    const std::pair<unsigned int, unsigned int> &face_range) const
+  {
+    FEFaceEvaluation<dim, fe_degree, n_q_points_1d, dim, Number> fe_eval(data,
+                                                                         true);
+
+    // OBS! This will need to be modified once the Piola transform is
+    // implemented
+    unsigned int n_cells =
+      data.get_dof_handler().get_triangulation().n_active_cells();
+    Number piola =
+      (dim == 2) ? n_cells : Utilities::pow((int)std::cbrt(n_cells), 4);
+
+    for (unsigned int face = face_range.first; face < face_range.second; ++face)
+      {
+        fe_eval.reinit(face);
+        fe_eval.gather_evaluate(src,
+                                EvaluationFlags::values |
+                                  EvaluationFlags::gradients);
```

Comment:
This is a good idea! However, I think that after the Piola transform is implemented (hopefully successful), I will be able to remove a lot of this test and for instance use `tests/matrix_free/matrix_vector_mf.h` since it is the same, just without the Piola scaling. Also feel like it would be easier to write an entirely new test instead? What do you think?